Changed to list comprehension

haru-44 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed
Change to list comprehension to improve readability.

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

haru-44 committed:
Changed to list comprehension

sylee957 commented on sympy/simplify/radsimp.py:
```diff
@@ -246,10 +246,9 @@ def parse_derivative(deriv):
-                if s != s0:
-                    raise NotImplementedError(
-                        'Improve MV Derivative support in collect')
+            if any(s != s0 for s in expr.variables):
```
```suggestion
            if s0 not in expr.variables:
```

sylee957 commented on sympy/concrete/delta.py:
```diff
@@ -100,9 +100,8 @@ def _has_simple_delta(expr, index):
-            for arg in expr.args:
-                if _has_simple_delta(arg, index):
-                    return True
+            if any(_has_simple_delta(arg, index) for arg in expr.args):
+                return True
```
```suggestion
            return any(_has_simple_delta(arg, index) for arg in expr.args)
```

smichr committed:
Update sympy/concrete/delta.py

Although this was consistent with the original behavior, since the if-block has a False fallback, it does not hurt to simplify this line and just return a False here and now.

Co-authored-by: S.Y. Lee  <sylee957@gmail.com>
